### ISSUE TITLE
Stringable false positive for string usage

### ISSUE THREAD

**Issue by dereuromark** [vimeo/psalm#9204]
> src/Database/Log/QueryLogger.php:55:29: PossiblyInvalidArgument: Argument 2 of Cake\Log\Log::write expects string, but possibly different type Stringable|string provided (see https://psalm.dev/092)
```php
    public function log($level, string|\Stringable $message, array $context = []);
```

The code inside this log() method is:
```php
        if (...) {
            $message = 'connection={connection} duration={took} rows={numRows} ' . $message;
        }
        Log::write('debug', $message, $context);
```
which is
```php
    public static function write($level, string $message, $context = []): bool
```
$message can be string|Stringable, so the auto cast is always fine.
Thus this is a false positive.

**Comment by dereuromark** [comment]
https://psalm.dev/r/f7677f5bcc
Since the `.=` concat is only happening in some cases, this kind of clear implicit cast is covered only for those
But passing this into a method with type string should also be fine.

**Comment by orklah** [comment]
https://3v4l.org/ns478

Stringable is not a valid input for a string param. What saves you is that PHP performs an implicit __toString call when you're not in strict_types=1. This is exactly what the error tells you. Feel free to suppress this issue if you feel strongly otherwise

**Comment by orklah** [comment]
Well, there's a false negative though. For your snippet, the line 4 should also have emitted an ImplicitToStringCast when concatenating with a string

**Comment by dereuromark** [comment]
Actually, no, string casting works for some reason
PHP ...^^

https://3v4l.org/U72sM

**Comment by orklah** [comment]
Yeah it works, but the ImplicitToStringCast doesn't make a difference between what works and what doesn't: https://psalm.dev/docs/running_psalm/issues/ImplicitToStringCast/

So the issue should be raised anyway in theory

**Comment by dereuromark** [comment]
in theory maybe, in practice only the type method param is an issue here. Thx, we will fix it on our side now.
